tests/functional/event_handler/_pydantic/test_openapi_validation_middleware.py

```diff
@@ -1983,3 +2259,131 @@ def get_user(user_id: int) -> UserModel:
     assert response_body["name"] == "User123"
     assert response_body["age"] == 143
     assert response_body["email"] == "user123@example.com"
+
+
+def test_validate_pydantic_query_params_with_config_dict_and_validators(gw_event):
+    """Test that Pydantic models with ConfigDict, aliases, and validators work correctly"""
+    from typing import Any
+
+    from pydantic import UUID4, AfterValidator, Base64UrlStr, ConfigDict, StringConstraints, alias_generators
+
+    del gw_event["multiValueHeaders"]
+    del gw_event["multiValueQueryStringParameters"]
+
+    app = APIGatewayRestResolver(enable_validation=True)
+    app.enable_swagger(path="/swagger")
+
+    def _validate_powertools(value: str) -> str:
+        if not value.startswith("Powertools"):
+            raise ValueError("Full name must start with 'Powertools'")
+        return value
+
+    class QuerySimple(BaseModel):
+        full_name: Annotated[str, StringConstraints(min_length=5), AfterValidator(_validate_powertools)]
+        next_token: Base64UrlStr
+        search_id: str
+
+    @app.get("/query-model-simple")
+    def query_model(params: Annotated[QuerySimple, Query()]) -> dict[str, Any]:
+        return {
+            "fullName": params.full_name,
+            "nextToken": params.next_token,
+            "searchId": params.search_id,
+        }
+
+    class QueryAdvanced(BaseModel):
+        full_name: Annotated[str, StringConstraints(min_length=5)]
+        next_token: str
+        search_id: Annotated[str, Field(alias="id")]  # Using str instead of UUID4 for simpler testing
+
+        model_config = ConfigDict(
+            alias_generator=alias_generators.to_camel,
+            validate_by_alias=True,
+            serialize_by_alias=True,
+        )
```

Comment:
This is the same for `populate_by_alias` and is prefered for pydantic >2.11.

```suggestion
            alias_generator=alias_generators.to_camel,
            validate_by_alias=True,
            validate_by_name=True,
            serialize_by_alias=True,
        )
```